Jenkinsfile

```diff
@@ -0,0 +1,121 @@
+                slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
+                postCommentIfPR("Internal build has been started, your results will be available at build completion.", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
+            }
+        }
+
+        // Checkout the repository
+        stage('Checkout repo') {
+            steps {
+                retry(3) {
+                    checkout([$class: 'GitSCM', branches: [[name: "refs/heads/${BRANCH_NAME}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'WipeWorkspace'], [$class: 'LocalBranch', localBranch: "${BRANCH_NAME}"], [$class: 'CloneOption', timeout: 30, shallow: true]], submoduleCfg: [], userRemoteConfigs: scm.userRemoteConfigs ])
+                }
+            }
+        }
+
+        stage('Full Build except itest') {
+            when {
+                expression { env.CHANGE_ID == null } 
+            }
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withMaven(maven: 'maven-latest', globalMavenSettingsConfig:  '51e52749-c47a-4e11-9c58-0adf485626f5', mavenSettingsConfig: 'feca3f61-1da1-4887-a9ad-dd4a41fd4423', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                    sh 'mvn -P!documentation -P!findbugs -DskipTests=true -Dpmd.skip=true -Djacoco.skip=true -Dfindbugs.skip=true -Dcheckstyle.skip=true clean install'
+                }
+            }
+        }
+
+        stage('Full Build') {
+            when {
+                expression { env.CHANGE_ID == null } 
+            }
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withMaven(maven: 'maven-latest', mavenSettingsConfig: 'feca3f61-1da1-4887-a9ad-dd4a41fd4423', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                    sh 'mvn clean install -T 1C'
+                }
+            }
+        }
+
+        /*
+          Deploy stage will only be executed for deployable branches. 
+          It will also only deploy in the presence of an environment variable JENKINS_ENV = 'prod'. This can be passed in globally from the jenkins master node settings.
+        */
+        stage('Deploy') {
+            when {
+                allOf {
+                    expression { env.CHANGE_ID == null }
+                    expression { env.BRANCH_NAME ==~ /((?:\d*\.)?\d*\.x|master)/ }
+                    environment name: 'JENKINS_ENV', value: 'prod'
+                }
+            }
+            steps{
+                withMaven(maven: 'maven-latest', jdk: 'jdk8-latest', globalMavenSettingsConfig: '51e52749-c47a-4e11-9c58-0adf485626f5', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                    sh 'mvn deploy -nsu -DskipTests=true -Dpmd.skip=true -Dfindbugs.skip=true -Dcheckstyle.skip=true -Djacoco.skip=true'
```

Comment:
You can remove some of these flags: documentation (profile doesn't exist), project doesn't use pmd, findbugs, checkstyle.

Reply:
Removed flags